### PR TITLE
handle potential errors in quality encodings of Fastqs

### PR DESCRIPTION
**Pull request by pmb59**
- Add initial Fastq checks with `bbmap` and `seqtk` before running IRAP
- Fix out-of-range quality values instead of crashing with a warning
- edit the README file

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -360,7 +360,7 @@ rule run_irap:
             echo "irap_single_lib SE finished for {wildcards.sample}"
         else
             # fastq is PE
-            reformat.sh ow=t int=t vpair=t vint=t in={params.root_dir}/{input.fastq} out1=$workingDir/${{localFastqPath}}_1.fastq out2=$workingDir/${{localFastqPath}}_2.fastq
+            reformat.sh ow=t ibq=t int=t vpair=t vint=t in={params.root_dir}/{input.fastq} out1=$workingDir/${{localFastqPath}}_1.fastq out2=$workingDir/${{localFastqPath}}_2.fastq
```
Comment (anilthanki) [inline Snakefile:363]: Something for future:

`vint` sets `vpair` to true and `interleaved` or `int` to true. so we dont need `int=t vpair=t vint=t`

**Review by anilthanki — APPROVED** [review]
LGTM

File: Snakefile (hunk 1 above)
Comment (anilthanki) [reply Snakefile:363]: https://github.com/BioInfoTools/BBMap/blob/a9ceda047a7c918dc090de0fdbf6f924292d4a1f/sh/reformat.sh#L41

**Review by irisdianauy — APPROVED** [review]
LGTM

**Comment by anilthanki** [comment]
LGTM

**Pull request merged**